modules/instrument_manager/jsx/uploadForm.js

```diff
@@ -0,0 +1,97 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+
+class InstrumentUploadForm extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: {},
+      selectedFile: null,
+    };
+
+    this.fileSelected = this.fileSelected.bind(this);
+    this.upload = this.upload.bind(this);
+  }
+
+  fileSelected(element, file) {
+    this.setState({
+      selectedFile: file,
+    });
+  }
+
+  upload() {
+    const data = new FormData();
+    data.append('install_file', this.state.selectedFile);
+
+    fetch(this.props.action, {
+      method: 'POST',
+      credentials: 'same-origin',
+      body: data,
+    })
+    .then((resp) => {
+      if (resp.status == 201) {
+        swal({
```

Comment:
We can now use the `swal2` dependency which uses promise syntax (e.g. `then`, `catch) which I find a lot more intuitive than the bootstrap swal. Perhaps it's not accessible on major yet though.

Reply:
maybe in a second PR